src/e2e/java/teammates/e2e/pageobjects/StudentFeedbackResultsPage.java

```diff
@@ -0,0 +1,564 @@
+        additionalInfo = appendOptions(additionalInfo, questionDetails.getConstSumOptions());
+        additionalInfo.append(System.lineSeparator());
+        if (questionDetails.isPointsPerOption()) {
+            additionalInfo.append("Points per option: ");
+        } else {
+            additionalInfo.append("Total points: ");
+        }
+        additionalInfo.append(questionDetails.getPoints());
+        return additionalInfo.toString();
+    }
+
+    private String getConstSumRecipientsAddInfo(FeedbackConstantSumQuestionDetails questionDetails) {
+        StringBuilder additionalInfo = new StringBuilder("Distribute points (among recipients) question");
+        additionalInfo.append(System.lineSeparator());
+        if (questionDetails.isPointsPerOption()) {
+            additionalInfo.append("Points per recipient: ");
+        } else {
+            additionalInfo.append("Total points: ");
+        }
+        additionalInfo.append(questionDetails.getPoints());
+        return additionalInfo.toString();
+    }
+
+    private StringBuilder appendOptions(StringBuilder info, List<String> options) {
+        StringBuilder additionalInfo = info;
+        for (String option : options) {
+            additionalInfo.append(option).append(System.lineSeparator());
+        }
+        return additionalInfo.deleteCharAt(additionalInfo.length() - 1);
+    }
+
+    private WebElement getQuestionResponsesSection(int questionNum) {
+        return browser.driver.findElement(By.id("question-" + questionNum + "-responses"));
+    }
+
+    private void showAdditionalInfo(int qnNumber) {
+        WebElement additionalInfoLink = getQuestionResponsesSection(qnNumber).findElement(By.id("additional-info-link"));
+        if (additionalInfoLink.getText().equals("[more]")) {
+            click(additionalInfoLink);
+            waitUntilAnimationFinish();
+        }
+    }
+
+    private String getAdditionalInfo(int questionNum) {
+        showAdditionalInfo(questionNum);
+        return getQuestionResponsesSection(questionNum).findElement(By.id("additional-info")).getText();
+    }
+
+    private WebElement getGivenResponseField(int questionNum, String receiver) {
+        int recipientIndex = getGivenRecipientIndex(questionNum, receiver);
+        return getQuestionResponsesSection(questionNum)
+                .findElements(By.cssSelector("#given-responses tm-single-response"))
+                .get(recipientIndex);
+    }
+
+    private int getGivenRecipientIndex(int questionNum, String recipient) {
+        List<WebElement> recipients = getQuestionResponsesSection(questionNum)
+                .findElements(By.cssSelector("#given-responses #response-recipient"));
+        for (int i = 0; i < recipients.size(); i++) {
+            if (recipients.get(i).getText().split("To: ")[1].equals(recipient)) {
+                return i;
+            }
+        }
+        throw new RuntimeException("Recipient not found: " + recipient);
```

Comment:
All the other `throw new RuntimeException` should be changed to `Assert.fail` as well

Reply:
Hmm... For some of them if I change to using `fail`, I have to add a `return null` after to avoid missing return statement error. Is it better if I change to `throw new AssertionError()` instead?